vpn/ipc/outbound.go

```diff
@@ -0,0 +1,159 @@
+package ipc
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	runtimeDebug "runtime/debug"
+	"time"
+
+	"github.com/sagernet/sing-box/adapter"
+	"github.com/sagernet/sing-box/common/conntrack"
+	"github.com/sagernet/sing-box/protocol/group"
+	"github.com/sagernet/sing/service"
+)
+
+type selection struct {
+	GroupTag    string `json:"groupTag"`
+	OutboundTag string `json:"outboundTag"`
+}
+
+// SelectOutbound selects an outbound within a group.
+func SelectOutbound(groupTag, outboundTag string) error {
+	_, err := sendRequest[empty]("POST", selectEndpoint, selection{groupTag, outboundTag})
+	return err
+}
+
+func (s *Server) selectHandler(w http.ResponseWriter, r *http.Request) {
+	if s.service.Status() != StatusRunning {
+		http.Error(w, "service not ready", http.StatusServiceUnavailable)
+		return
+	}
+	var p selection
+	err := json.NewDecoder(r.Body).Decode(&p)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	outbound, err := getGroupOutbound(s.service.Ctx(), p.GroupTag)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	selector, isSelector := outbound.(*group.Selector)
```

Comment:
If possible, it would be better if we could check if it contains the function that we want based on a interface (here we only need the struct to have the function `SelectOutbound(tag string) bool`). In this way we won't need to compare if it's a sing-box selector or sbx